### ISSUE TITLE
json setTo with array lookup fails on attempt to match an email address-type pattern

### ISSUE THREAD

**Issue by ewan-chalmers** [rangav/thunder-client-support#1141]
**Describe the bug**
json setTo fails when array lookup attempts to match an email address-type pattern.

**To Reproduce**

I want to make a full match for an email address:
```
json.Resources[userName=ewan.chalmers@foo.bar.com].id
```

in a response like:
```
{
  "totalResults": 1,
  "schemas": [
    "urn:ietf:params:scim:api:messages:2.0:ListResponse"
  ],
  "Resources": [
    {
      "active": true,
      "id": "123ABC456",
      "userName": "ewan.chalmers@foo.bar.com"
    }
  ]
}
```

result:
```
Set Env Failed: Unable to parse left input query - json.Resources[userName=ewan.chalmers@foo.bar.com].id
```

It's not what I need, but I am able to get a successful match up to the first `. ` following the `@`,  e.g. this works
```
json.Resources[userName=^ewan.chalmers@foo].id
```

The specific pattern I originally tried to match was in fact email address  + `@` + suffix, e.g. 
```
  "userName": "ewan.chalmers@foo.bar.com@www.foo.bar"
```
My specific case is not important, but I mention this to suggest that a fix should not depend on allowing a standard email pattern in match expressions.

FWIW, I tried using `urlencode` but this did not resolve the issue (`json.Resources[userName={{VAR | urlEncode}}].id`).

**Expected behavior**
I can successfully execute a test with json setTo, matching an email-like pattern in an attribute value of an array object.

**Platform:** 
 - OS:  macOS
 - vscode version: 1.77.3
 - extension version: 2.6.1 

**Your Team Size Using TC:**

Unknown

**Comment by rangav** [comment]
Thanks @ewan-chalmers for reporting the bug, Will look into it.

similar to https://github.com/rangav/thunder-client-support/issues/1123

**Comment by ewan-chalmers** [comment]
Similarly, don't think we can have json setTo array match where property key contains dots and/or is a quoted string

e.g. don't think we can make a match expression for `Resources["urn:ietf:params:scim:schemas:extension:ibm:2.0:User". unqualifiedUserName]`

```
{
  "totalResults": 1,
  "schemas": [
    "urn:ietf:params:scim:api:messages:2.0:ListResponse"
  ],
  "Resources": [
    {
      "emails": [
      ],
      "meta": {
      },
      "name": {
      },
      "urn:ietf:params:scim:schemas:extension:ibm:2.0:User": {
        "lastLogin": "2023-05-03T13:59:31Z",
        "realm": "www.foo.bar.com",
        "unqualifiedUserName": "ewan.chalmers@foo.bar.com",
      },
      "active": true,
      "id": "123ABC456",
      "userName": "ewan.chalmers@foo.bar.com@www.foo.bar"
    }
  ]
}
```

**Comment by rangav** [comment]
Alternatively you can also use Custom scripting

https://github.com/rangav/thunder-client-support/blob/master/docs/filters.md#custom-filters

**Comment by rangav** [comment]
You can use filters to get the desired results

`json.Resources | filter(userName=ewan.chalmers@foo.bar.com) | prop("id")`

See docs
https://github.com/rangav/thunder-client-support/blob/master/docs/filters.md#built-in-filters

**Comment by rangav** [comment]
I am closing this issue, feel free to reopen if needed

**Comment by rangav** [comment]
re-opened this issue, as the bug is not fixed yet, but will fix it in the next update.

**Comment by rangav** [comment]
This bug is fixed now and published to the marketplace, please update to `v2.8.0`

See all features released
https://github.com/rangav/thunder-client-support/releases/tag/v2.8.0

Please let me know your feedback.
